File: iris-backend/app/schemas/release.py
"""Software release schema."""
from marshmallow import Schema, fields


class SoftwareReleaseSchema(Schema):
    release_id = fields.Int(dump_only=True)
    product_id = fields.Int(allow_none=True)
    product_name = fields.Str(required=True)
    domain = fields.Str(allow_none=True)
    category = fields.Str(allow_none=True)
    new_version = fields.Str(required=True)
    release_date = fields.Date(required=True)
    release_title = fields.Str(required=True)
    description = fields.Str(allow_none=True)
    highlights = fields.Raw(allow_none=True)
    match_criteria = fields.Raw(required=True)
    is_active = fields.Bool(load_default=True)
    added_by = fields.Int(dump_only=True, allow_none=True)
    created_at = fields.DateTime(dump_only=True)
    updated_at = fields.DateTime(dump_only=True)

File: iris-frontend/src/types/release.ts
export interface Release {
  release_id: number;
  product_id?: number;
  product_name: string;
  domain: string;
  new_version: string;
  release_title: string;
  description?: string;
  highlights: string[];
  release_date: string;
  is_active: boolean;
  match_criteria?: Record<string, unknown>;
  matched_accounts_count?: number;
  estimated_opportunity?: number;
  created_at: string;
}

export interface ReleaseMatch {
  match_id: number;
  release_id: number;
  account_id: number;
  account_name?: string;
  install_id: number;
  installed_version?: string;
  match_reason?: string;
  match_score: number;
  reminder_status: 'Pending' | 'Reminded' | 'Closed';
  created_at: string;
}

export interface ReleaseFilters {
  domain?: string;
  is_active?: boolean;
  page?: number;
  per_page?: number;
}
